infra/prover-cluster/components/ApiGatewayComponent.ts

```diff
@@ -94,14 +127,20 @@ export class ApiGatewayComponent extends BaseComponent {
         });
 
         // Create ALB listener for HTTP (port 80) on internal ALB
+        // Use replaceOnChanges to force replacement when target group changes
+        // This is necessary because AWS won't allow updating a listener to use a different target group
         new aws.lb.Listener("boundless-internal-alb-listener", {
             loadBalancerArn: this.internalAlb.arn,
             port: 80,
             protocol: "HTTP",
             defaultActions: [{
                 type: "forward",
-                targetGroupArn: this.targetGroup.arn,
+                targetGroupArn: this.internalTargetGroup.arn,
             }],
+        }, {
+            dependsOn: [this.internalTargetGroup],
+            replaceOnChanges: ["defaultActions"],
```

Comment:
Replacing the listener can cause some amount of downtime, is that acceptable here?